onnxruntime/core/framework/node_index_info.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/node_index_info.h"
+
+#include "core/framework/mlvalue_name_idx_map.h"
+#include "core/graph/graph_viewer.h"
+#include "core/graph/node_arg.h"
+
+namespace onnxruntime {
+
+NodeIndexInfo::NodeIndexInfo(const GraphViewer& graph_viewer, const MLValueNameIdxMap& mlvalue_idx_map)
+    : max_mlvalue_idx_{mlvalue_idx_map.MaxIdx()} {
+  std::size_t total_def_count{};
+
+  bool include_missing_optional_defs = true;
+
+  for (const auto& node : graph_viewer.Nodes()) {
+    node.ForEachDef([&](const onnxruntime::NodeArg& /*arg*/, bool /*is_input*/) {
+      ++total_def_count;
+    },
+                    include_missing_optional_defs);
+  }
+
+  // init all to -1
+  node_offsets_.resize(graph_viewer.MaxNodeIndex(), -1);
+  node_values_.resize(total_def_count, -1);
+  int cur_idx = 0;
+
+  for (auto& node : graph_viewer.Nodes()) {
+    node_offsets_[node.Index()] = cur_idx;
+
+    node.ForEachDef([&](const onnxruntime::NodeArg& node_arg, bool /*is_input*/) {
+      auto& name = node_arg.Name();
+      if (node_arg.Exists()) {
+        int index;
+        Status status = mlvalue_idx_map.GetIdx(name, index);
+        ORT_ENFORCE(status.IsOK(), status.ErrorMessage());
+        node_values_[cur_idx] = index;
+      }
+      // else it's a missing optional input or output so leave the -1
+
+      ++cur_idx;
+    },
+                    include_missing_optional_defs);
```

Comment:
>                     [](start = 0, length = 20)

indent here seems confusing. Maybe try .clangformat?

Reply:
That is clangformat. The indent is aligning with the opening '(' of the ForEachDef call. The lambda which is the first argument has different indent rules which is why it looks weird. 

---
In reply to: [252393442](https://github.com/Microsoft/onnxruntime/pull/410#discussion_r252393442) [](ancestors = 252393442)

Reply:
Maybe you can move the lambda in ForEachDef in a new line to make it align better for reading.

---
In reply to: [252437805](https://github.com/Microsoft/onnxruntime/pull/410#discussion_r252437805) [](ancestors = 252437805,252393442)